From foo@baz Tue Apr  9 12:12:43 2002
Date: Tue, 18 Jul 2006 10:59:59 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: PCI: enable driver multi-threaded probe


This turns on multhreaded probe for all PCI drivers.  It can cause bad
problems on multi-processor machines that take a while to find their
root disks.  Use at
your own risk right now.


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pci-driver.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/pci/pci-driver.c
+++ gregkh-2.6/drivers/pci/pci-driver.c
@@ -422,6 +422,7 @@ int __pci_register_driver(struct pci_dri
 	drv->driver.bus = &pci_bus_type;
 	drv->driver.owner = owner;
 	drv->driver.kobj.ktype = &pci_driver_kobj_type;
+	drv->driver.multithread_probe = 1;
 
 	spin_lock_init(&drv->dynids.lock);
 	INIT_LIST_HEAD(&drv->dynids.list);
